syncserver/src/lib.rs

```diff
@@ -3,12 +3,243 @@
+            .wrap(syncstorage_web::middleware::WeaveTimestamp::default())
+            .wrap(middleware::sentry::SentryWrapper::<
+                syncstorage_web::ServerState,
+                syncstorage_web::ApiError,
+            >::default())
+            .wrap(middleware::rejectua::RejectUA::<syncstorage_web::ServerState>::default())
+            // Followed by the "official middleware" so they run first.
+            // actix is getting increasingly tighter about CORS headers. Our server is
+            // not a huge risk but does deliver XHR JSON content.
+            // For now, let's be permissive and use NGINX (the wrapping server)
+            // for finer grained specification.
+            .wrap(build_cors($settings))
+            .wrap(syncstorage_web::middleware::EmitTokenserverOrigin::default())
+            .configure(
+                syncstorage_web::get_configurator(
+                    &$settings.syncstorage,
+                    $settings.statsd_host.as_deref(),
+                    $settings.statsd_port,
+                )
+                .expect("failed to build syncstorage configurator"),
+            )
+            .configure(configure_dockerflow)
+    };
+}
+
+#[cfg(all(not(feature = "syncstorage"), feature = "tokenserver"))]
+#[macro_export]
+macro_rules! build_app {
+    ($settings: expr) => {
+        App::new()
+            .data(Arc::new($settings.master_secret.clone()))
+            // Middleware is applied LIFO
+            // These will wrap all outbound responses with matching status codes.
+            .wrap(middleware::sentry::SentryWrapper::<
+                tokenserver_web::ServerState,
+                tokenserver_web::ApiError,
+            >::default())
+            .wrap(tokenserver_web::middleware::LoggingWrapper::new())
+            .wrap(middleware::rejectua::RejectUA::<tokenserver_web::ServerState>::default())
+            // Followed by the "official middleware" so they run first.
+            // actix is getting increasingly tighter about CORS headers. Our server is
+            // not a huge risk but does deliver XHR JSON content.
+            // For now, let's be permissive and use NGINX (the wrapping server)
+            // for finer grained specification.
+            .wrap(build_cors($settings))
+            .configure(
+                tokenserver_web::get_configurator(
+                    &$settings.tokenserver,
+                    $settings.statsd_host.as_deref(),
+                    $settings.statsd_port,
+                )
+                .expect("failed to build tokenserver configurator"),
+            )
+            .configure(configure_dockerflow)
+    };
+}
+
+#[cfg(all(feature = "syncstorage", feature = "tokenserver"))]
+#[macro_export]
+macro_rules! build_app {
+    ($settings: expr, $blocking_threadpool: expr) => {
+        App::new()
+            .data(Arc::new($settings.master_secret.clone()))
+            .service(
+                web::scope("/token")
```

Comment:
I include the same prefixes that exist for syncserver here https://github.com/mozilla-services/syncserver/blob/c116417d653432db9470098236fb117adb349c36/syncserver/__init__.py#L158-L159